packages/components/src/components/ConvertTokenInput/ConvertTokenInput.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  Input,
+  InputGroup,
+  InputRightElement,
+  InputLeftAddon,
+  Button,
+  InputProps,
+} from '@chakra-ui/react';
+import React, { useState } from 'react';
+
+export interface ConvertTokenInputProps {
+  /**
+   * The value to be converted
+   */
+  value: string;
+}
+
+// A number input component with built-in conversion feature from wei <-> ether
+export const ConvertTokenInput: React.FC<ConvertTokenInputProps & InputProps> = ({
+  value,
+  ...props
+}) => {
+  const [currentUnit, setCurrentUnit] = useState('ether');
+  const [targetUnit, setTargetUnit] = useState('wei');
+  const [val, setVal] = useState(value);
+
+  // The convert() function converts the current value into the unit selected by the user
+  const convert = () => {
+    if (targetUnit === 'wei') {
+      setCurrentUnit('wei');
+      setTargetUnit('ether');
+      setVal((parseFloat(val) * 1000000000000000000).toFixed(2));
+    } else if (targetUnit === 'ether') {
+      setCurrentUnit('ether');
+      setTargetUnit('wei');
+      setVal((parseFloat(val) / 1000000000000000000).toFixed(18));
+    }
+  };
```

Comment:
Maybe we could move this to a custom hook? I think it's something we might reuse in other places as well. @Dhaiwat10 @etr2460

Reply:
We definitely can @christiananese , but this functionality is already provided by [web3js](https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#fromwei) and [ether-converter package](https://www.npmjs.com/package/ether-converter) . Will it be a good idea to use these libraries?